task1/src/main/kotlin/jub/task1/game/PageSearch.kt

```diff
@@ -13,10 +18,52 @@ data class SearchPath(
 class PageSearch(
     private val finalPage: String = KOTLIN_PAGE,
 ) {
-    fun search(startPage: String, searchDepth: Int, threadsCount: Int): SearchPath = TODO("Not implemented yet")
+    fun restoreAnswer(startPage: String, previousLink: MutableMap<String, String>): SearchPath {
+        var currentLink = finalPage
+        val path = mutableListOf(currentLink)
+        while (currentLink != startPage) {
+            currentLink = previousLink[currentLink] ?: KEY_NOT_FOUND
+            if (currentLink == KEY_NOT_FOUND) {
+                return SearchPath(NOT_FOUND, listOf())
+            }
+            path.add(currentLink)
+        }
+        path.reverse()
+        return SearchPath(path.size - 1, path)
+    }
+
+    fun search(startPage: String, searchDepth: Int, threadsCount: Int): SearchPath = runBlocking {
+        val customDispatcher = Executors.newFixedThreadPool(threadsCount).asCoroutineDispatcher()
+        val coroutineScope = CoroutineScope(customDispatcher)
+        val queue: ArrayDeque<String> = ArrayDeque()
```

Comment:
```suggestion
        val queue = ArrayDeque<String>()
```
JFYI, it's not necessary to duplicate the type of a val, but it's up to you.